beta-src/src/utils/drawArrow.ts

```diff
@@ -0,0 +1,52 @@
+export default function drawArrow(
+  /**
+   * color code passed from enum ArrowColors based on move type
+   */
+  arrowColor: string,
+  /**
+   * source is the id of the element the arrow begins at
+   */
+  source: string,
+  /**
+   * The resulting object after selecting the full map using D3 (NOT A SVG ELEMENT); must be able to target centers and defs
+   */
+  d3MapSelector: any,
+  /**
+   * target is the id of the element the arrow ends at
+   */
+  target: string,
+) {
+  const arrowIdNumber = `${source}-${target}`;
+
+  const sourceNodeData = (
+    d3MapSelector.select(`#${source}`).node() as SVGSVGElement
+  ).getBBox();
+  const targetNodeData = (
+    d3MapSelector.select(`#${target}`).node() as SVGSVGElement
+  ).getBBox();
+
+  d3MapSelector
+    .select("defs")
+    .append("marker")
+    .attr("id", `arrowhead__${arrowIdNumber}`)
+    .attr("markerWidth", 8)
+    .attr("markerHeight", 8)
+    .attr("refX", 0)
+    .attr("refY", 4)
+    .attr("orient", "auto")
+    .append("polygon")
+    .attr("points", "0 0, 8 4, 0 8")
+    .attr("fill", arrowColor);
+
+  d3MapSelector
```

Comment:
Why not just pass in the SVG element and then use the d3 selector in here like they are used in WDMapController so that types are inferred?

Reply:
This would require me to import d3 into the function file which I feel to be pointless if the controller of the draw arrow functionality will more than likely need to have d3 and have a reference to the MapControllers already declared d3 selected svg; this way we can just pass whats already declared.